pandas/core/frame.py

```diff
@@ -7453,7 +7453,11 @@ def _dispatch_frame_op(self, right, func: Callable, axis: AxisInt | None = None)
             #  fails in cases with empty columns reached via
             #  _frame_arith_method_with_reindex
 
-            # TODO operate_blockwise expects a manager of the same type
+            if isinstance(self._mgr, BlockManager) and isinstance(
```

Comment:
can this be done inside one of the Manager methods?

Reply:
done